File: app/src/main/java/com/prueba/mytodolist/addTask/AddTaskViewModel.java
package com.prueba.mytodolist.addTask;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.prueba.mytodolist.database.TaskEntry;
import com.prueba.mytodolist.repository.TaskRepository;

public class AddTaskViewModel extends ViewModel {

    private LiveData<TaskEntry> task;

    public AddTaskViewModel(int taskId, Application application) {
        TaskRepository taskRepository = new TaskRepository(application);
        task = taskRepository.getTaskById(taskId);
    }

    public LiveData<TaskEntry> getTask() {
        return task;
    }
}

File: app/src/main/java/com/prueba/mytodolist/addTask/AddTaskViewModel.kt
package com.prueba.mytodolist.addTask

import android.app.Application
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.prueba.mytodolist.database.TaskEntry
import com.prueba.mytodolist.repository.TaskRepository

class AddTaskViewModel(taskId: Int, application: Application): ViewModel() {

    var task: LiveData<TaskEntry> = TaskRepository(application).getTaskById(taskId)
}